src/components/interactive-builder/modals/question/question-form/common/sortable-tag/sortable-tag.component.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import classNames from 'classnames';
+import { useSortable } from '@dnd-kit/sortable';
+import { CSS } from '@dnd-kit/utilities';
+import { useTranslation } from 'react-i18next';
+import { IconButton } from '@carbon/react';
+import { Draggable, WarningAltFilled, TrashCan } from '@carbon/react/icons';
+import styles from './sortable-tag.scss';
+
+interface SortableTagProps {
+  id: string;
+  text: string;
+  onDelete?: () => void;
+  isInvalid?: boolean;
+}
+
+export const SortableTag: React.FC<SortableTagProps> = ({ id, text, onDelete, isInvalid }) => {
+  const { t } = useTranslation();
+  const { attributes, listeners, setNodeRef, transform, transition, isDragging } = useSortable({
+    id,
+  });
+
+  const dynamicStyle = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  };
+
+  return (
+    <div
+      ref={setNodeRef}
+      style={dynamicStyle}
+      className={classNames(styles.sortableTagWrapper, {
```

Comment:
Prefer to add the invalid styles if invalid here as well so that the whole container gets highlighted and not just the text.